app/controllers/reports_controller.rb

```diff
@@ -47,7 +47,7 @@ class ReportsController < ApplicationController
     :step_contents,
     :result_contents
   ]
-  before_action :check_destroy_permissions, only: :destroy
+  before_action :check_manage_permissions, only: %i(edit update destroy)
```

Comment:
Well, one way would be to throw the check for `report_ids` completely out from permissions checking, because we don't do it in other places anyway, so this is an exception and hence it doens't play nice with the new permissions structure. the checks for parameters are usually done in the actual method, for which we checked the permissions.

Reply:
You'll know better if we break something, but I suggest to do the following:
* rename `check_create_permissions` into `check_manage_permissions`:
```ruby
def check_manage_permissions
  render_403 unless can_manage_reports?(@project)
end
```
* add `check_report_ids` or a similar check, that only triggers on `only: %i(edit update destroy)`:
```ruby
def check_report_ids
    render_404 unless params.include? :report_ids
  end
```